src/main/java/com/auth0/RequestProcessor.java

```diff
@@ -83,73 +84,82 @@ AuthorizeUrl buildAuthorizeUrl(HttpServletRequest request, String redirectUri, S
         if (responseTypeList.contains(KEY_TOKEN) || responseTypeList.contains(KEY_ID_TOKEN)) {
             creator.withParameter(KEY_RESPONSE_MODE, KEY_FORM_POST);
         }
+        if (verifyOptions.getMaxAge() != null) {
+            creator.withParameter(KEY_MAX_AGE, verifyOptions.getMaxAge().toString());
+        }
         return creator;
     }
 
     /**
      * Entrypoint for HTTP request
      * <p>
-     * 1). Responsible for validating the request and ensuring the state value in session storage matches the state value passed to this endpoint.
-     * 2). Exchanging the authorization code received with this HTTP request for auth0 tokens.
-     * 3). Getting the user information associated to the id_token/access_token.
-     * 4). Storing both tokens and user information into session storage.
-     * 5). Clearing the stored state value.
-     * 6). Handling success and any failure outcomes.
+     * 1). Responsible for validating the request.
+     * 2). Exchanging the authorization code received with this HTTP request for Auth0 tokens.
+     * 3). Validating the ID Token.
+     * 4). Clearing the stored state, nonce and max_age values.
+     * 5). Handling success and any failure outcomes.
      *
      * @throws IdentityVerificationException if an error occurred while processing the request
      */
     Tokens process(HttpServletRequest req) throws IdentityVerificationException {
         assertNoError(req);
         assertValidState(req);
 
-        Tokens tokens = tokensFromRequest(req);
+        Tokens frontChannelTokens = getFrontChannelTokens(req);
+        Tokens codeExchangeTokens = frontChannelTokens;
+        List<String> responseTypeList = getResponseType();
+
+        if (responseTypeList.contains(KEY_ID_TOKEN) && frontChannelTokens.getIdToken() == null) {
+            throw new InvalidRequestException(MISSING_ID_TOKEN, "Id Token is missing from the response.");
+        }
+        if (responseTypeList.contains(KEY_TOKEN) && frontChannelTokens.getAccessToken() == null) {
```

Comment:
I assume you say that because using implicit grant flows with form-post is as safe as using code grant flows when we want to save that extra code-exchange network request. This has been like this for some time now. If we are going to do such change or deprecation I suggest we discuss this first and start a project to update the rest of the SDKs as well.

Reply:
I understand if this is backwards-compat but, regardless of the response mode, the response type should not include `token`